### ISSUE TITLE
Carousel is broken

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#2661]
@jhpedemonte 

Add a Carousel to a View.  The styling appears to be missing.  There are no icons, only a black bar and arrow buttons. 

Try changing the data in the input and you get this error:

```
Uncaught Error: Use StoreCarousel or DataCarousel instead of Carousel. lib/dojo/dojox/mobile/Carousel.js:55
_3.startup lib/dojo/dojox/mobile/Carousel.js:55
require.cache.dijit/_Container._a4.addChild lib/dojo/dijit/dijit.js:15
_2b03._addChildHooked :50010/maqetta/app/davinci/davinci.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:62378
_2b03.addChild :50010/maqetta/app/davinci/davinci.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:62355
_2800.execute :50010/maqetta/app/davinci/davinci.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:57583
_1e42.execute :50010/maqetta/app/davinci/davinci.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:41698
ret.withDoc dojo.js:6382
_1c56.execute :50010/maqetta/app/davinci/davinci.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:38837
_2a7._executeCompoundCommand DataStoreBasedWidgetInput.js:3401
_2a7._getDummyDataUpdateWidgetCommand DataStoreBasedWidgetInput.js:3428
_c7 dojo.js:15
_36 dojo.js:15
_16 dojo.js:15
req dojo.js:15
_2a7._getDummyDataUpdateWidgetCommand DataStoreBasedWidgetInput.js:3419
_2a7.getUpdateWidgetCommand DataStoreBasedWidgetInput.js:3372
_2a7.onOk DataStoreBasedWidgetInput.js:3367
lang.hitch dojo.js:4885
_53a.(anonymous function)._53f dojo.js:6771
lang.extend._onBlur :50010/maqetta/app/dijit/dijit.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:2151
_231._setStack :50010/maqetta/app/dijit/dijit.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:2378
_231._onTouchNode :50010/maqetta/app/dijit/dijit.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:2345
_231.registerWin._241 :50010/maqetta/app/dijit/dijit.js?b76c656a27d79da7b363f3c0e1cb4a25deb1f226:2250
```

**Comment by peller** [comment]
...or, the merry go round broke down

**Comment by doronrosenberg** [comment]
@jhpedemonte @JonFerraiolo @peller 

In dojo 1.8, carousel no longer supports stores - instead there is a DataCarousel and a StoreCarousel, which are separate widgets.  The easiest solution would be to remove Carousel and replace it with StoreCarousel, the helpers should migrate over easily and probably with no changes.

Are we planning to switching to using data rather than stores? If so this would be blocked by a bug for creating a base Data class like we have for stores (DataStoreBasedWidgetInput)

**Comment by JonFerraiolo** [comment]
> Are we planning to switching to using data rather than stores?

Maybe we can talk about this issue on the next team phone call. 

@billreed63 @aerwin @peller

**Comment by doronrosenberg** [comment]
@JonFerraiolo @jhpedemonte 

So here is a question - do we keep Carousel as the name in the widget box but switch to using DataCarousel under the covers?  Or do we add DataCarousel as a separate widget?

Also, I assume this is a major migration issue, as m6 maqetta created carousels won't work in m7 because of the dojo 1.8 change.

**Comment by peller** [comment]
Given that this is dojox.mobile's API breakage, I don't think we should attempt to migrate it.  IIUC, Carousel is a base class now and not something that works OOTB?  Do we want to replace it with DataCarousel?  I suppose we should use the name DataCarousel, since we will someday also offer StoreCarousel.

**Comment by doronrosenberg** [comment]
Carousel can contain CarouselItem now.  So we could change Carousel to build those using a smart input, and then have a separate DataCarousel that works with datastores.

**Comment by JonFerraiolo** [comment]
I wouldn't worry much about backwards compatibility with Carousel widgets in old user documents.

If Dojo offers a plain Carousel widget with CarouselItem, then I would suggest having separate Carousel (, CarouselItem maybe at least under the hood), vs DataCarousel and StoreCarousel. That seems to be the pattern we are adopted (e.g., EdgeToEdge vs EdgeToEdgeDataList).

But another question is what is DataCarousel vs StoreCarousel? Is DataCarousel using the old dojo.data stuff? Shouldn't we avoid proliferating that and move forward to dojo.store?

**Comment by doronrosenberg** [comment]
@JonFerraiolo we could switch to dojo.store, but that would require more work as we would need to write new helpers/etc.  Probably 3-4 days of work to write a new Carousel widget and a new StoreCarousel with store-specific base class and helpers.

**Comment by JonFerraiolo** [comment]
Because DataCarousel sounds like a relative short task, I guess we should stick with DataCarousel for time being and see if we have time to get to StoreCarousel (and then think about dropping DataCarousel)

**Comment by doronrosenberg** [comment]
fixed by d0a5c209bc806369b85807691963b2b285280143
